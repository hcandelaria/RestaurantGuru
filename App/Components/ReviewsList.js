//flow
import React, { Component } from 'react'
import { Content, Text, Card, CardItem, Body, Button, Icon, Left, Container} from 'native-base'
import { connect } from 'react-redux'

export default class CategoryList extends Component {
  render () {
    return (
      <Content>
        {
          this.props.reviews
            ? this.props.reviews.map((review, index) =>

              <Container
                key={index}
              >
                <Left>
                  <Text>{` ${review.user.name}: ${review.review_text}`}</Text>
                  <Text>
                    <Icon
                      small
                      name='thumbs-up'
                      />
                    {` ${review.likes} ${review.rating}/5` }
                  </Text>
                </Left>
              </Container>
            )
          : <Text>Reviews are not available at this moment.</Text>

        }
      </Content>
    )
  }
}
