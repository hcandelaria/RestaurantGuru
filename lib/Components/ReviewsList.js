var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));var _createClass2=_interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _possibleConstructorReturn2=_interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));var _getPrototypeOf2=_interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));var _inherits2=_interopRequireDefault(require("@babel/runtime/helpers/inherits"));var _react=_interopRequireWildcard(require("react"));var _nativeBase=require("native-base");var _reactRedux=require("react-redux");var _jsxFileName="C:\\Users\\Smilodon\\Interviews\\bin2rid\\RestaurantGuru\\App\\Components\\ReviewsList.js";var CategoryList=function(_Component){(0,_inherits2.default)(CategoryList,_Component);function CategoryList(){(0,_classCallCheck2.default)(this,CategoryList);return(0,_possibleConstructorReturn2.default)(this,(0,_getPrototypeOf2.default)(CategoryList).apply(this,arguments));}(0,_createClass2.default)(CategoryList,[{key:"render",value:function render(){return _react.default.createElement(_nativeBase.Content,{__source:{fileName:_jsxFileName,lineNumber:9}},this.props.reviews?this.props.reviews.map(function(review,index){return _react.default.createElement(_nativeBase.Container,{key:index,__source:{fileName:_jsxFileName,lineNumber:14}},_react.default.createElement(_nativeBase.Left,{__source:{fileName:_jsxFileName,lineNumber:17}},_react.default.createElement(_nativeBase.Text,{__source:{fileName:_jsxFileName,lineNumber:18}}," "+review.user.name+": "+review.review_text),_react.default.createElement(_nativeBase.Text,{__source:{fileName:_jsxFileName,lineNumber:19}},_react.default.createElement(_nativeBase.Icon,{small:true,name:"thumbs-up",__source:{fileName:_jsxFileName,lineNumber:20}})," "+review.likes+" "+review.rating+"/5")));}):_react.default.createElement(_nativeBase.Text,{__source:{fileName:_jsxFileName,lineNumber:29}},"Reviews are not available at this moment."));}}]);return CategoryList;}(_react.Component);exports.default=CategoryList;