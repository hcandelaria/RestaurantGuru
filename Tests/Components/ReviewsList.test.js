import 'react-native'
import React from 'react'
import ReviewsList from '../../App/Components/ReviewsList'
import renderer from 'react-test-renderer'

import reviews from '../../App/Fixtures/reviews.json'

test('ReviewsList component renders correctly with reviews', () => {
  const tree = renderer.create(
    <ReviewsList
      reviews={reviews.all_reviews}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
