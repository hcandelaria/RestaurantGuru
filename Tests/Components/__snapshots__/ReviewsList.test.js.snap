// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList component renders correctly with reviews 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "height": 1334,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
           John Doe: The best latte I've ever had. It tasted a little sweet
        </Text>
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          <Text
            allowFontScaling={false}
            small={true}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï‰–
          </Text>
           0 5/5
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
